src/main/java/com/amazonaws/encryptionsdk/kmsv2/AwsKmsMrkAwareMasterKeyProvider.java

```diff
@@ -134,54 +126,19 @@ private RuntimeException clientSupplierComboException() {
               + "withCredentials or withClientBuilder");
     }
 
-    /**
-     * Configures the {@link AwsKmsMrkAwareMasterKeyProvider} to use specific credentials. If a
-     * builder was previously set, this will override whatever credentials it set.
-     *
-     * @see KmsMasterKeyProvider.Builder#withCredentials(AWSCredentialsProvider)
-     */
-    public Builder withCredentials(
-        AWSCredentialsProvider credentialsProvider) {
-      if (regionalClientSupplier_.isPresent()) {
-        throw clientSupplierComboException();
-      }
-
-      if (templateBuilder_ == null) {
-        templateBuilder_ = AWSKMSClientBuilder.standard();
-      }
-
-      templateBuilder_.setCredentials(credentialsProvider);
-
-      return this;
-    }
-
-    /**
-     * Configures the {@link AwsKmsMrkAwareMasterKeyProvider} to use specific credentials. If a
-     * builder was previously set, this will override whatever credentials it set.
-     *
-     * @see KmsMasterKeyProvider.Builder#withCredentials(AWSCredentials)
-     */
-    public Builder withCredentials(AWSCredentials credentials) {
-      return withCredentials(new AWSStaticCredentialsProvider(credentials));
-    }
-
     /**
      * Configures the {@link AwsKmsMrkAwareMasterKeyProvider} to use settings from this {@link
-     * AWSKMSClientBuilder} to configure KMS clients. Note that the region set on this builder will
-     * be ignored, but all other settings will be propagated into the regional clients.
-     *
-     * <p>This method will overwrite any credentials set using {@link
-     * #withCredentials(AWSCredentialsProvider)}.
+     * KmsClientBuilder} to configure KMS clients. Note that the region set on this builder will be
+     * ignored, but all other settings will be propagated into the regional clients.
      *
-     * @see KmsMasterKeyProvider.Builder#withClientBuilder(AWSKMSClientBuilder)
+     * @see KmsMasterKeyProvider.Builder#builderSupplier(Supplier)
      */
-    public Builder withClientBuilder(AWSKMSClientBuilder builder) {
-      if (regionalClientSupplier_.isPresent()) {
+    public Builder builderSupplier(Supplier<KmsClientBuilder> supplier) {
```

Comment:
I'm still not sure if this needs to be a supplier, but if it does, should we provide a convenience method that will also just accept a KmsClientBuilder to make this simpler for customers?

Reply:
It needs to be a supplier because we cannot clone the builders. If the user does not provide something here. It defaults to `KmsClient.builder()`. The can also specify this explicitly with: `.builderSupplier(KmsClient::builder)` and I intend to illustrate this through docs.